src/hooks/core/chat/useFetchChannelHook.ts

```diff
@@ -90,6 +106,9 @@ const useFetchChannelHook = () => {
       await Promise.all(
         (channel?.messages || []).map(async (message) => {
           if (message?.type === 'deleted') return;
+          if (message?.type === 'system') {
+            message = helperFollowingSystemMessage(message);
+          }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

Similar to the previous comment, avoid mutating the `message` object directly. Use the updated `helperFollowingSystemMessage` function that returns a new object instead.
```diff
-           if (message?.type === 'system') {
-             message = helperFollowingSystemMessage(message);
-           }
+           if (message?.type === 'system') {
+             message = {...helperFollowingSystemMessage(message)};
+           }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->